Utils/auto_update_docker/extract_affected_content_items.py

```diff
@@ -0,0 +1,447 @@
+import math
+from packaging.version import Version
+import logging
+from collections import defaultdict
+from pathlib import Path
+from typing import Any
+import os
+import typer
+from dotenv import load_dotenv
+from neo4j import Transaction
+
+from demisto_sdk.commands.common.handlers import DEFAULT_JSON_HANDLER as json
+from demisto_sdk.commands.content_graph.interface.neo4j.neo4j_graph import (
+    Neo4jContentGraphInterface as ContentGraphInterface,
+)
+from demisto_sdk.commands.common.docker.docker_image import DockerImage
+
+logging.basicConfig(level=logging.INFO)
+load_dotenv()
+app = typer.Typer(no_args_is_help=True)
+CWD = os.getcwd()
+
+
+def load_json(path: str) -> dict[str, Any]:
+    with open(path, encoding="utf-8") as f:
+        return json.loads(f.read())
+
+
+def filter_content_items_to_run_on(
+    batch_config: dict[str, Any],
+    content_items_coverage: dict[str, int],
+    content_items_by_docker_image: list[dict[str, Any]],
+    target_docker_tag: str,
+    nightly_packs: list[str],
+    docker_image: str,
+    benchmark_docker_tags: dict[str, str]
+) -> dict[str, Any]:
+    """Collect the content items with respect to the batch config.
+
+    Args:
+        batch_config (dict[str, Any]): The batch config. (Default or custom)
+        content_items_coverage (dict[str, int]): Coverage of content items.
+        content_items_by_docker_image (list[dict[str, Any]]): A list of content items per docker image to check if they fit
+        into the current batch.
+        target_docker_tag (str): The target docker tag.
+        nightly_packs (list[str]): The nightly packs.
+
+    Returns:
+        dict[str, Any]: A dictionary where the key is the docker image, and the values are
+        the affected content items, pr tags, and target tag for a specific docker image.
+    """
+    affected_content_items: list[str] = []
+    only_nightly: bool = batch_config.get("only_nightly", False)
+    min_cov: int = batch_config["min"]
+    max_cov: int = batch_config["max"]
+    logging.info(f"{min_cov} - {max_cov} in filter_content_items_to_run_on")
+    support_levels = batch_config.get("support", [])
+    for content_item in content_items_by_docker_image:
+        content_item_path = Path(content_item["content_item"])
+        content_item_support = content_item["support_level"]
+        content_item_pack_path = content_item["pack_path"]
+        content_item_docker_image_tag = content_item["docker_image_tag"]
+
+        if only_nightly and content_item_pack_path not in nightly_packs:
+            logging.info(f"Pack path {content_item_pack_path} for {content_item} is not in nightly, skipping.")
+            continue
+
+        if docker_image in benchmark_docker_tags:
+            docker_image_tag_benchmark = benchmark_docker_tags[docker_image]
+            if Version(content_item_docker_image_tag) > Version(docker_image_tag_benchmark):
```

Comment:
this isnt a great check actually, because there are times where for some reason we might want to downgrade a library in an image.
The "source of truth" for what is a later docker image should be the int after the last index of .
For eg. 1.2.3.4554 vs 1.1.1.5555 -> 1.1.1.5555 should be considered larger